### PR TITLE
Chore/test additions

### PR DESCRIPTION
**Pull request by michaelrob**
* Added specs for stories and users controller
* Added factories for stories and users
* Removed "skip_before_action :authenticate_user_from_token!" from story controller, we should be authenticating a user on all their requests :)

### CODE REVIEW

File: README.rdoc [hunk 1]
```diff
@@ -27,7 +27,7 @@ API docs to come.
 
 -------------------------------
 
-*Contribution*
+*Contributing*
```
Comment (nizmow) [inline README.rdoc]: 👍

File: spec/controllers/v1/stories_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+      before do
+        params[:story][:body] = nil
+
+        post :create, params
+      end
+
+      it 'should return story_create_error json' do
+        expect(response).to_not have_http_status :success
+        expect(response.body).to include 'translation missing: en.story_create_error'
```
Comment (nizmow) [inline spec/controllers/v1/stories_controller_spec.rb]: Well I guess we should add translations sometime, but when we do we can update these specs. :)

Comment (michaelrob) [inline spec/controllers/v1/stories_controller_spec.rb]: Thats very true :) I'm sure we can make some much more helpful messages.

File: spec/controllers/v1/stories_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      end
+    end
+  end
+
+  describe '#index' do
+    let!(:authorization) { request.headers['Authorization'] = user.access_token }
+
+    context 'authorized' do
+      it 'should return all stories for user' do
+
```
Comment (nizmow) [inline spec/controllers/v1/stories_controller_spec.rb]: Putting simply `pending` here is best practice to let us know the test is still to come.

File: spec/controllers/v1/stories_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      end
+    end
+
+    context 'not authorized' do
+
```
Comment (nizmow) [inline spec/controllers/v1/stories_controller_spec.rb]: And here.

File: spec/controllers/v1/stories_controller_spec.rb (hunk 3 above)
Comment (michaelrob) [inline spec/controllers/v1/stories_controller_spec.rb]: Fair :)

File: spec/controllers/v1/stories_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    context 'authorized' do
+      let!(:authorization) { request.headers['Authorization'] = user.access_token }
+
+      before do
+        get :show, id: story.id
+      end
+
+      it 'should return specific story for user' do
+        # not 100% sure on these expects. There must be a better way for us to check this
```
Comment (nizmow) [inline spec/controllers/v1/stories_controller_spec.rb]: You can check for the presence of instance variables, and check the content of instance variables. I found a fairly modern blog post about roughly what to do here: http://tarynsauer.tumblr.com/post/76908014101/things-i-learned-about-rspec-this-week

I'm sure we do this in Bodypass. :)

File: spec/controllers/v1/stories_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      end
+    end
+
+    context 'not authorized' do
+      before do
+        get :show, id: story.id
+      end
+
+      it 'should return unauthorized error' do
+        expect(response.body).to include 'translation missing: en.application_controller.unauthorized'
```
Comment (nizmow) [inline spec/controllers/v1/stories_controller_spec.rb]: You could test that you return 401 or whatever, too.

File: spec/controllers/v1/users_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+          password: 'password',
+          password_confirmation: 'password'
+        }
+      }
+    end
+
+    context 'user successful' do
+      it 'should create new user and return success' do
+        expect { post :create, params }.to change { User.count }.by(1)
+        expect(response.body).to include 'access_token'
```
Comment (nizmow) [inline spec/controllers/v1/users_controller_spec.rb]: Oh, just realised you're possibly not using instance variables for these API requests...

**Comment by nizmow** [comment]
Looks good! Just a few comments but without digging through the rest of the code I'm not sure how relevant they all are. :)

**Comment by michaelrob** [comment]
Thanks :) I'll have a good look and see how I go with them.

File: spec/controllers/v1/stories_controller_spec.rb (hunk 5 above)
Comment (michaelrob) [inline spec/controllers/v1/stories_controller_spec.rb]: Yeah I considered doing that, you mentioned below in your other comment.... we're not using instance variables here in the controllers :)

**Pull request merged**